src/app.js

```diff
@@ -62,18 +61,99 @@ go.app = function() {
                 characters_per_page: 160,
                 options_per_page : 3,
                 next: function(choice) {
-                    return 'states:end';
+                    return { 
+                        name: 'states:retrieve-location',
+                        creator_opts : {
+                            full_geoid : choice.value,
+                            full_name : choice.label
+                        }
+                    };
                 }   
             });
         });
-      
+
+        self.states.add('states:retrieve-location', function(name, opts) {
+            return self
+                .http.get('http://wazimap.co.za/profiles/' + opts.full_geoid + '.json')
+                .then(function(response) {
+                    opts.data = response.data;
+                    return  self.states.create(
+                        'states:select-section', opts);
+                });
+        });
+
+        self.states.add('states:select-section', function(name, opts) {
+            return new ChoiceState(name, {
+                question: 'I would like to query:',
+                choices: [
+                    new Choice('elections', 'Elections'),
+                    new Choice('demographics', 'Demographics'),
+                    new Choice('households', 'Households'),
+                    new Choice('service_delivery', 'Service Delivery'),
+                    new Choice('economics', 'Economics'),
+                    new Choice('education', 'Education'),
+                    new Choice('children', 'Children'),
+                    new Choice('child_households', 'Child-headed Households')
+                        ],
+                next: function(choice) {
+                    return {
+                        name: 'states:display-data',
+                        creator_opts : {
+                            section_name : choice.label,
+                            section_id : choice.value,
+                            opts_data : opts.data,
+                            location_name : opts.full_name
+                        }
+                    };
+                }
+            });
+        });
+
+        self.states.add('states:display-data', function(name, opts) {
+            var section_data = opts.opts_data[opts.section_id]; 
+            var sub_section_data = _.map(section_data, function(d) {
+                return ('Registered Voters: ' + d.registered_voters.values.this);
+            });
```

Comment:
What's happening here is that `elections` itself has multiple sub-sections (e.g. `provincial_2014`), so you're getting one result per-subsection. I think let's leave this for now and sort it out when we figure out the data formatting in the next PR?